src/api/login.go

```diff
@@ -22,10 +25,20 @@ func Login(p *dbtool.Pool) gin.HandlerFunc {
 			return
 		}
 
-		// Save access token and resfresh token [Testing for now]
+		v := validator.New()
+		if valErr := v.Struct(userCred); valErr != nil {
+			c.JSON(
+				http.StatusBadRequest,
+				&errorResponse{
+					Msg: "Authentication failed. Invalid data type.",
+				},
+			)
+			return
+		}
+
 		var user dbtool.User
-		dbQuery := p.Where("email = ?", userCred.Email).Find(&user)
-		if dbQuery.Error != nil {
+		dbErr := p.Query(&user, map[string]interface{}{"email": userCred.Email})
+		if dbErr != nil {
```

Comment:
그냥 err해도 될듯. 밑과 같이 해도 되지만, 이런 쿼리가 길어지는게 문제라면 쿼리를 다른 함수로 뺴도 될듯 재사용성 있으면

```suggestion
		if err:= p.Query(&user, map[string]interface{}{"email": userCred.Email}); err != nil {
```